deeplite/profiler/formatter.py

```diff
@@ -1,22 +1,52 @@
 from collections import OrderedDict
 import logging
-import sys
 
 from .metrics import Comparative, compare_status_values, Metric
 
 
-def getLogger():
-    nroot = logging.RootLogger(logging.DEBUG)
-    consol_level = logging.DEBUG
+class _LoggerHolder:
+    """
+    Holds the logger for the profiler module and redirect any calls made to its held logger. If none is held
+    then defaults to logging.getLogger('deeplite_profiler'). This allows customization of the logging
+    being used in the profiler by an external library by importing setLogger.
+    """
+    __singleton = False
+
+    def __new__(cls, *args, **kwargs):
+        if cls.__singleton:
+            raise TypeError("This class is a singleton!")
+        cls.__singleton = True
+        return super().__new__(cls)
+
+    def __init__(self):
+        self._logger = None
+
+    @property
+    def logger(self):
+        if self._logger is None:
+            self._logger = logging.getLogger('deeplite_profiler')
+            self._logger.debug("Defaulting to logging.getLogger('deeplite_profiler')'s logger")
+        return self._logger
+
+    @logger.setter
+    def logger(self, new_logger):
+        self._logger = new_logger
+        self._logger.debug("Profiler's logger set to {}".format(new_logger))
+
+    def __getattribute__(self, item):
```

Comment:
that's a bit twisted, I think the logger_holder should stay just that, a holder.. and getLogger should return its inside _logger

Reply:
I added a docstring to explain the intention. This is to be able to dynamically change it.